elastic/kibana
# [Discover] Fix filter data getting cleared when filter doesn't match active data view

## Summary


When the user attempted to edit a filter that was created with a data view other than the active one, the filter data inputs were empty except for the raw query DSL.  This PR fixes this behavior

**Testing**
1. Create filter in Discover
2. Switch data view
3. Edit filter
4. This should work

Fixes #154785


### Checklist
